uvtools/dspec.py

```diff
@@ -150,9 +155,24 @@ def high_pass_fourier_filter(data, wgts, filter_size, real_delta, clean2d=False,
         # run clean
         if dndim == 1:
             # For 1D data array run once
-            _d_cl, info = aipy.deconv.clean(_d, _w, area=area, tol=tol, stop_if_div=False, maxiter=maxiter, gain=gain)
-            _d_res = info['res']
-            del info['res']
+            if not linear:
+                _d_cl, info = aipy.deconv.clean(_d, _w, area=area, tol=tol, stop_if_div=False, maxiter=maxiter, gain=gain)
+                _d_res = info['res']
+                del info['res']
+            else:
+                ff = [ tol ]
+                if isinstance(filtersize, np.float) == 1:
```

Comment:
this should just be `if isinstance(filtersize, float):`

Reply:
yeah. I don't know how that happened... Changed.